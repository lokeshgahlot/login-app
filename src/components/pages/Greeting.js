import React from 'react';
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {Jumbotron, Button, ButtonToolbar} from "react-bootstrap";
import Spinner from "../Spinner";
import {logout} from "../../actions/logout.actions";
import {fetchUser} from "../../actions/fetchUser.actions"

export const GreetingPage = ({name, userId, logout, fetchUser}) => {
  if (!name) {
    fetchUser && fetchUser(userId);
    return (
      <div className="col-xs-offset-5">
        <Spinner size="64px"/>
      </div>
    )
  }
  return (
    <Jumbotron>
      <h1>Hello, {name}!</h1>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. 
        Suspendisse tristique sapien vitae arcu dapibus, ut tempus nibh vestibulum. 
        In vitae ligula turpis. Nam purus tortor, lacinia quis nulla 
        vitae, sagittis lobortis neque. Donec posuere volutpat ex, 
        condimentum convallis magna rutrum elementum. Vivamus feugiat, 
        libero sed euismod sollicitudin, ipsum orci posuere sapien, at cursus 
        libero tortor vel turpis. Suspendisse potenti. 
        Maecenas blandit risus quis euismod rhoncus. 
        Morbi ac consequat nisi, eu vehicula dui. Quisque tincidunt
        bibendum auctor. Etiam a auctor lectus.
      </p>
      <Button bsStyle="primary" className= "logout-btn" onClick={logout}>Logout</Button>
    </Jumbotron>
  );
}

GreetingPage.propTypes = {
  name: PropTypes.string,
  nauserIdme: PropTypes.string,
  logout: PropTypes.func,
  fetchUser: PropTypes.func
};

export const mapSTP = (state) => ({
  name: state.get("user").get("name"),
  userId: state.get("authenticationInfo").get("id")
});

export default connect(mapSTP, {
  logout,
  fetchUser
})(GreetingPage);