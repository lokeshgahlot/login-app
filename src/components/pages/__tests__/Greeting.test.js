import React from "react";
import {shallow} from "enzyme"
import {GreetingPage, mapSTP} from "../Greeting";
import {fromJS} from "immutable";

describe("<GreetingPage />", () => {
  it ("renders spinner", () => {
    const wrapper = shallow(<GreetingPage />);
    expect(wrapper).toMatchSnapshot();
  });

  it ("calls fetchUser()", () => {
    const fn = jest.fn();
    const wrapper = shallow(<GreetingPage fetchUser={fn} userId="1234"/>);
    expect(fn.mock.calls.length).toBe(1);
    expect(fn.mock.calls[0][0]).toBe("1234");
  });

  it ("renders greeting page", () => { 
    const wrapper = shallow(<GreetingPage name="Joe"/>);
    expect(wrapper).toMatchSnapshot();
  });

  it ("renders logout()", () => { 
    const fn = jest.fn();
    const wrapper = shallow(<GreetingPage name="Joe" logout={fn}/>);
    wrapper.find(".logout-btn").simulate("click");
    expect(fn.mock.calls.length).toBe(1);
  });

  it("mapSTP()",()=> {
    const state = fromJS({
      user: {
        name: "Joe"
      }, 
      authenticationInfo: {
        id: "1234"
      }
    });
    expect(mapSTP(state)).toEqual({name: "Joe", userId: "1234"});
  });

});