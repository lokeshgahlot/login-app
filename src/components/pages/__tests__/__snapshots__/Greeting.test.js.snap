// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GreetingPage /> renders greeting page 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GreetingPage
    name="Joe"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bsClass": "jumbotron",
      "children": Array [
        <h1>
          Hello, 
          Joe
          !
        </h1>,
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse tristique sapien vitae arcu dapibus, ut tempus nibh vestibulum. In vitae ligula turpis. Nam purus tortor, lacinia quis nulla vitae, sagittis lobortis neque. Donec posuere volutpat ex, condimentum convallis magna rutrum elementum. Vivamus feugiat, libero sed euismod sollicitudin, ipsum orci posuere sapien, at cursus libero tortor vel turpis. Suspendisse potenti. Maecenas blandit risus quis euismod rhoncus. Morbi ac consequat nisi, eu vehicula dui. Quisque tincidunt bibendum auctor. Etiam a auctor lectus.
        </p>,
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="primary"
          className="logout-btn"
          disabled={false}
          onClick={undefined}
        >
          Logout
        </Button>,
      ],
      "componentClass": "div",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Hello, ",
            "Joe",
            "!",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Hello, ",
          "Joe",
          "!",
        ],
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse tristique sapien vitae arcu dapibus, ut tempus nibh vestibulum. In vitae ligula turpis. Nam purus tortor, lacinia quis nulla vitae, sagittis lobortis neque. Donec posuere volutpat ex, condimentum convallis magna rutrum elementum. Vivamus feugiat, libero sed euismod sollicitudin, ipsum orci posuere sapien, at cursus libero tortor vel turpis. Suspendisse potenti. Maecenas blandit risus quis euismod rhoncus. Morbi ac consequat nisi, eu vehicula dui. Quisque tincidunt bibendum auctor. Etiam a auctor lectus.",
        },
        "ref": null,
        "rendered": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse tristique sapien vitae arcu dapibus, ut tempus nibh vestibulum. In vitae ligula turpis. Nam purus tortor, lacinia quis nulla vitae, sagittis lobortis neque. Donec posuere volutpat ex, condimentum convallis magna rutrum elementum. Vivamus feugiat, libero sed euismod sollicitudin, ipsum orci posuere sapien, at cursus libero tortor vel turpis. Suspendisse potenti. Maecenas blandit risus quis euismod rhoncus. Morbi ac consequat nisi, eu vehicula dui. Quisque tincidunt bibendum auctor. Etiam a auctor lectus.",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "active": false,
          "block": false,
          "bsClass": "btn",
          "bsStyle": "primary",
          "children": "Logout",
          "className": "logout-btn",
          "disabled": false,
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "Logout",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsClass": "jumbotron",
        "children": Array [
          <h1>
            Hello, 
            Joe
            !
          </h1>,
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse tristique sapien vitae arcu dapibus, ut tempus nibh vestibulum. In vitae ligula turpis. Nam purus tortor, lacinia quis nulla vitae, sagittis lobortis neque. Donec posuere volutpat ex, condimentum convallis magna rutrum elementum. Vivamus feugiat, libero sed euismod sollicitudin, ipsum orci posuere sapien, at cursus libero tortor vel turpis. Suspendisse potenti. Maecenas blandit risus quis euismod rhoncus. Morbi ac consequat nisi, eu vehicula dui. Quisque tincidunt bibendum auctor. Etiam a auctor lectus.
          </p>,
          <Button
            active={false}
            block={false}
            bsClass="btn"
            bsStyle="primary"
            className="logout-btn"
            disabled={false}
            onClick={undefined}
          >
            Logout
          </Button>,
        ],
        "componentClass": "div",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Hello, ",
              "Joe",
              "!",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Hello, ",
            "Joe",
            "!",
          ],
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse tristique sapien vitae arcu dapibus, ut tempus nibh vestibulum. In vitae ligula turpis. Nam purus tortor, lacinia quis nulla vitae, sagittis lobortis neque. Donec posuere volutpat ex, condimentum convallis magna rutrum elementum. Vivamus feugiat, libero sed euismod sollicitudin, ipsum orci posuere sapien, at cursus libero tortor vel turpis. Suspendisse potenti. Maecenas blandit risus quis euismod rhoncus. Morbi ac consequat nisi, eu vehicula dui. Quisque tincidunt bibendum auctor. Etiam a auctor lectus.",
          },
          "ref": null,
          "rendered": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse tristique sapien vitae arcu dapibus, ut tempus nibh vestibulum. In vitae ligula turpis. Nam purus tortor, lacinia quis nulla vitae, sagittis lobortis neque. Donec posuere volutpat ex, condimentum convallis magna rutrum elementum. Vivamus feugiat, libero sed euismod sollicitudin, ipsum orci posuere sapien, at cursus libero tortor vel turpis. Suspendisse potenti. Maecenas blandit risus quis euismod rhoncus. Morbi ac consequat nisi, eu vehicula dui. Quisque tincidunt bibendum auctor. Etiam a auctor lectus.",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": false,
            "block": false,
            "bsClass": "btn",
            "bsStyle": "primary",
            "children": "Logout",
            "className": "logout-btn",
            "disabled": false,
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "Logout",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<GreetingPage /> renders spinner 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GreetingPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Spinner
        color="#000000"
        size="64px"
      />,
      "className": "col-xs-offset-5",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "color": "#000000",
        "size": "64px",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Spinner
          color="#000000"
          size="64px"
        />,
        "className": "col-xs-offset-5",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "color": "#000000",
          "size": "64px",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
